doc/man/scons.xml

```diff
@@ -3981,20 +3981,20 @@ extern "C"
 char function_name(void);
 </programlisting>
 
+<para>
+      The optional <parameter>funcargs</parameter> can be defined to specify an argument list for the generated
+      function invocation.
+</para>
+
 <para>
 Note: if <parameter>header</parameter> is supplied,
-it should <emphasis>not</emphasis>
-include the standard header file that declares
-<parameter>function_name</parameter>,
-and it <emphasis>should</emphasis> include a
-dummy prototype similar to the default case.
-Compilers reject builds where a function call does
-not match the declared prototype as happens
-if the "real" header is included,
-and modern compilers are now rejecting
-implicit function declarations.
+  and the function_name being called takes any parameters, appropriate values for those
+  function arguments should be supplied
+  by using the <parameter>funcargs</parameter>.
```

Comment:
Here's a possible alternative (I think it's okay as is, I was just thinking of how I would have explained this... This would combine the two paragraphs into the new single para, and retains some of what was in the note originally.)
```xml
<para>
If <parameter>header</parameter> is supplied,
it should <emphasis>not</emphasis>
include the standard header file that declares
<parameter>function_name</parameter>,
and it <emphasis>should</emphasis> include a
dummy prototype similar to the default case.
If this is not possible,
the optional <parameter>funcargs</parameter>
argument can be used to specify a string containing
an argument list with the same number of arguments as
in the prototype (they can just be 0 values or empty strings).
Modern C/C++ compilers reject implicit function declarations,
and may also reject function calls that don't look
like they match the protoype.
</para>
```

Reply:
I rather like this change.

Perhaps wordsmithing the following:
> can be used to specify a string containing
an argument list with the same number of arguments as
in the prototype (they can just be 0 values or empty strings).

To something along the lines of:
>can be used to specify a string containing
an argument list with the same number and type of arguments and as
in the prototype (they can simply be constant values of the correct type).

Reply:
> I rather like this change.
> 
> Perhaps wordsmithing the following:

that's fine by me (although apparently the types don't actually matter).

Reply:
> (although apparently the types don't actually matter)

I would nor have expected that.  If a prototype is present, I would have expected type checking (taking into account some types are automatically coerced) and possible compiler errors rather than linker errors, at least for c.

Reply:
Tests run on ubuntu/wsl2 and windows/pc.

Both of these tests will fail (passing a float where a char* is expected):
```
r = sconf.CheckFunc('strcpy', header='/* header */ char *strcpy(char *dest, char *src);', funcargs='0.0, ""')
assert r, "type failure prototype"

r = sconf.CheckFunc('strcpy', header='#include <string.h>', funcargs='0.0, ""')
assert r, "type failures include header"
```

However, both of these tests will pass (passing an int where a char* is expected):
```
r = sconf.CheckFunc('strcpy', header='/* header */ char *strcpy(char *dest, char *src);', funcargs='0, ""')
assert r, "type failure prototype"

r = sconf.CheckFunc('strcpy', header='#include <string.h>', funcargs='0, ""')
assert r, "type failures include header"
```

Which is what I now suspect you were alluding to in the statement above.

Zero can used/coerced into integers, floats, and (apparently) addresses which covers pointers and strings.

I wouldn't be surprised if there were warnings generated but am too lazy to test.

Perhaps the original text covers most all scenarios.